### ISSUE TITLE
plone-upgrade page should show warning when plone.app.upgrade is not available

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/Products.CMFPlone#3682]
Scenario:

- You have a Plone site, say 6.0.0b2.
- You upgrade the code to 6.0.0rc1.
- You do *not* include the `plone.app.upgrade` package.
- You start the site and go to `@@plone-upgrade`.
- This says your active configuration is 6008 and available is 6009.
- No upgrade steps are shown.
- You *do* see an upgrade button.
- Clicking the button does not really do much. At least the configuration will stay at the old 6008 version.

See [report by @1letter](https://community.plone.org/t/plone-6-0-0rc1-released/15885/7) and later [his solution](https://community.plone.org/t/plone-6-0-0rc1-released/15885/9). In the end the difference between the two situations he describes, is whether `plone.app.upgrade` is included or not (which gets included via the `Plone` egg, not the `Products.CMFPlone` egg). See that thread for screen shots.

I think we should check with `pkg_resources` if `plone.app.upgrade` is available. If not: show a warning. Maybe hide or disable the button.

BTW, this is no blocker in my opinion.

**Comment by yurj** [comment]
Isn't better to document when and how to include plone.app.upgrade? Shouldn't it be always included?

About the button, just check if there's at least 1 upgrade step to show it. Another problem is that Plone knows its profile version but need p.a.upgrade to really change it. So, why the profile hasn't (luckly) been upgraded clicking the button? Because the upgrade definition is in plone.app.upgrade.

Then why not just move the `@@plone-upgrade` view to `plone.app.upgrade`? This would solve the problem to see it even if p.a.upgrade is installed. This view is useless if plone.app.upgrade is not installed.

**Comment by jensens** [comment]
> Shouldn't it be always included?

No. I do not need it as ballast in some larger clusters for rendering instances except the one service instance. Also it would be a dependency indirection to depend on it in `Products.CMFPlone`. 

I like the idea to move the view to `plone.app.upgrade`, overriding one with a statement that there is no upgrade code installed in `Products.CMFPlone`.

**Comment by 1letter** [comment]
I like @jensens proposal. i think this is a good way.

**Comment by mauritsvanrees** [comment]
So have a `plone-upgrade` view in CMFPlone which only shows a message, and have the real `plone-upgrade` view in `plone.app.upgrade` and register it in an `overrides.zcml`? That feels a bit strange to me. Well, with a `zcml:condition="not-installed plone.app.upgrade"` on the original you would at least not need the `overrides.zcml`, making it a bit better.

Still, that would mean the [Upgrade view](https://github.com/plone/Products.CMFPlone/blob/6.0.0rc1/Products/CMFPlone/browser/admin.py#L311) would move to `plone.app.upgrade`, but the [`MigrationTool`](https://github.com/plone/Products.CMFPlone/blob/6.0.0rc1/Products/CMFPlone/MigrationTool.py#L128) would still stay in `CMFPlone`. I would rather keep them together. And we can't easily move the tool to `plone.app.upgrade`, because it is persistent. (It could probably be refactored to not be a persistent tool anymore, but that is a different story.)

**Comment by jensens** [comment]
Ah, the tool. All this old stuff in here. For now I'm fine with your approach. We can cleanup later.

**Comment by jensens** [comment]
Still relevant for Plone 6. plone-upgrade page should warn when plone.app.upgrade is not installed. Design agreed by maintainers, needs implementation.

**Comment by mauritsvanrees** [comment]
No, not relevant anymore.  I fixed this around Plone 6.0.0.

* Use `buildout.coredev` 6.2 and start with only `Products.CMFPlone`: `bin/instance-cmfplone fg`.
* Create a Plone site.
* Look at `@@plone-upgrade`:

<img width="780" height="90" alt="Image" src="https://github.com/user-attachments/assets/4be312e0-2224-4364-b3eb-f3c343f40059" />

This mentions nothing about a missing `plone.app.upgrade`, but this is a fresh site, so there are no upgrades to apply anyway, so I think this is fine.

* Now edit `Products/CMFPlone/profiles/default/metadata.xml` and increase its version number.
* Restart with the same `bin/instance-cmfplone fg`.
* Look at `@@plone-upgrade`:

<img width="779" height="669" alt="Image" src="https://github.com/user-attachments/assets/66f467ee-94e8-4f31-ae14-ca1913dd0836" />

It clearly says the `plone.app.upgrade` package is missing.
